classes/task/collect_metrics_task.php

```diff
@@ -0,0 +1,136 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace tool_cloudmetrics\task;
+
+use tool_cloudmetrics\metric;
+use tool_cloudmetrics\collector;
+
+/**
+ * Controls the running of metrics within a single task. Derive from scheduled_task to enable running from
+ * the moodle task system, but it is  intendend to be run separately.
```

Comment:
don't quite understand this comment?

Reply:
The comment has been modified. The original referenced something you said one, about possibly having a separate cron job to run this.